src/main/java/com/microsoft/azure/servicebus/jms/ServiceBusJmsConnectionFactory.java

```diff
@@ -134,6 +198,32 @@ private void initialize(String sasKeyName, String sasKey, String host, ServiceBu
 
             return properties;
         });
+        
+        
+		if (isAadAuthentication) {
+			
+			this.factory.setExtension(JmsConnectionExtensions.USERNAME_OVERRIDE.toString(), (connection, uri) -> {
+				return AAD_TOKEN_USERNAME;
+		    });
+		            
+			this.factory.setExtension(JmsConnectionExtensions.PASSWORD_OVERRIDE.toString(), (connection, uri) -> {
```

Comment:
What will happen if requesting a connection from this connection factory after the token expires?

Reply:
Since this is a function, so each connection created by the factory will apply this function. But since this implementation caches the aadToken, it won't try to retrieve a new one if the token expires after the first retrieval.

Reply:
Is address in Authentication getAadToken() we acquired a new token if expire